### PR TITLE
QSPARSE for BLR2 weakly admissible 1D laplace matrix.

### PR DESCRIPTION
**Pull request by v0dro**

### CODE REVIEW

File: examples/UMV_weak_1level_noblock.cpp [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+	fdiff = Hatrix::norm(A.U[i] * A.S(i, j) * A.V[j].transpose() - A.D(i, j));
+	diff += fdiff * fdiff;
+      }
+    }
+  }
+  return {A, std::sqrt(diff/norm)};
+}
+
+
+Hatrix::Matrix full_qr(Hatrix::Matrix& A) {
```
Comment (wawando) [inline examples/UMV_weak_1level_noblock.cpp:112]: The existing `Hatrix::qr` function also perform full QR if you pass Q and R matrices with the correct dimensions. You can simplify things here by creating a temporary matrix for R, call `Hatrix::qr` and return the Q

File: include/Hatrix/classes/Matrix.h [hunk 2]
```diff
@@ -57,6 +57,8 @@ class Matrix {
   size_t memory_used() const;
 
   size_t shared_memory_used() const;
+
+  Matrix transpose() const;
```
Comment (wawando) [inline include/Hatrix/classes/Matrix.h]: We already have a transpose function in `arithmetics.cpp` which also returns a `Matrix` object

Comment (v0dro) [reply include/Hatrix/classes/Matrix.h]: Removed.

File: examples/UMV_weak_1level_noblock.cpp (hunk 1 above)
Comment (v0dro) [reply examples/UMV_weak_1level_noblock.cpp:112]: This is a very bare bones implemetation so I've updated with your recommendation in the other file (`UMV_weak_1level.cpp`).

**Pull request merged**